Yosemite/Yosemite/Actions/ProductVariationAction.swift

```diff
@@ -28,4 +28,8 @@ public enum ProductVariationAction: Action {
     /// Requests the variations in a specified Order that have not been fetched.
     ///
     case requestMissingVariations(for: Order, onCompletion: (Error?) -> Void)
+
+    /// Delete an existing ProductVariation.
+    ///
+    case deleteProductVariation(productVariation: ProductVariation, onCompletion: (Result<ProductVariation, ProductUpdateError>) -> Void)
```

Comment:
Do you think it's better to have the result as `Result<Error, ProductUpdateError>`? as the deleted variation won't have any usage.

Reply:
That's actually a deeper question - why does API return deleted product and product variation? Since I don't know the answer - preferred to pass it as is 🙂 Also to keep consistency with existing "delete product" action.

In the end after I have everything connected in UI - I agree that there is no need to pass around value here.

I think using `Error` as a type for `Success` breaks logic of [Result type](https://developer.apple.com/documentation/swift/result). And it still needs some success value.

But I see `Result<Void, ProductUpdateError>` as a good fit, maybe you even meant this one 🙂

Reply:
>  why does API return deleted product and product variation?

I think that's just for some REST conventions.

> But I see Result<Void, ProductUpdateError> as a good fit, maybe you even meant this one

haha Yea, I totally meant that one, made a mistake while typing :-)